lib/charms/opensearch/v0/opensearch_distro.py

```diff
@@ -398,6 +406,43 @@ def _build_paths(self) -> Paths:
         """Build the Paths object."""
         pass
 
+    def apply_perf_templates_if_needed(self) -> bool:  # noqa: C901
```

Comment:
I would create a new module `performance_profile.py` where the comparison between current deployment-description profile and new config issued profile happens. 
Similar to the plugin manager